stl/inc/chrono

```diff
@@ -617,27 +669,1289 @@ namespace chrono {
+            const month _Temp{*this};
+            --(*this);
+            return _Temp;
+        }
+
+        constexpr month& operator+=(const months& _Months) noexcept;
+        constexpr month& operator-=(const months& _Months) noexcept;
+
+        _NODISCARD constexpr explicit operator unsigned int() const noexcept {
+            return static_cast<unsigned int>(_Month);
+        }
+        _NODISCARD constexpr bool ok() const noexcept {
+            return _Month >= 1 && _Month <= 12;
+        }
+
+    private:
+        unsigned char _Month;
+    };
+
+    _NODISCARD constexpr bool operator==(const month& _Left, const month& _Right) noexcept {
+        return static_cast<unsigned int>(_Left) == static_cast<unsigned int>(_Right);
+    }
+    _NODISCARD constexpr strong_ordering operator<=>(const month& _Left, const month& _Right) noexcept {
+        return static_cast<unsigned int>(_Left) <=> static_cast<unsigned int>(_Right);
+    }
+
+    _NODISCARD constexpr month operator+(const month& _Left, const months& _Right) noexcept {
+        const auto _Mo  = static_cast<long long>(static_cast<unsigned int>(_Left)) + (_Right.count() - 1);
+        const auto _Div = (_Mo >= 0 ? _Mo : _Mo - 11) / 12;
+        return month{static_cast<unsigned int>(_Mo - _Div * 12 + 1)};
+    }
+    _NODISCARD constexpr month operator+(const months& _Left, const month& _Right) noexcept {
+        return _Right + _Left;
+    }
+    _NODISCARD constexpr month operator-(const month& _Left, const months& _Right) noexcept {
+        return _Left + -_Right;
+    }
+    _NODISCARD constexpr months operator-(const month& _Left, const month& _Right) noexcept {
+        const auto _Mo = static_cast<unsigned int>(_Left) - static_cast<unsigned int>(_Right);
+        return months{_Mo <= 11 ? _Mo : _Mo + 12};
+    }
+
+    constexpr month& month::operator+=(const months& _Months) noexcept {
+        *this = *this + _Months;
+        return *this;
+    }
+    constexpr month& month::operator-=(const months& _Months) noexcept {
+        *this = *this - _Months;
+        return *this;
+    }
+
+    class year {
+    public:
+        year() = default;
+        constexpr explicit year(int _Val) noexcept : _Year{static_cast<short>(_Val)} {}
+
+        constexpr year& operator++() noexcept {
+            ++_Year;
+            return *this;
+        }
+        constexpr year operator++(int) noexcept {
+            return year{_Year++};
+        }
+        constexpr year& operator--() noexcept {
+            --_Year;
+            return *this;
+        }
+        constexpr year operator--(int) noexcept {
+            return year{_Year--};
+        }
+
+        constexpr year& operator+=(const years& _Years) noexcept {
+            _Year += static_cast<short>(_Years.count());
+            return *this;
+        }
+        constexpr year& operator-=(const years& _Years) noexcept {
+            _Year -= static_cast<short>(_Years.count());
+            return *this;
+        }

```

Comment:
We thought for a surprisingly long time about the correctness of this (which doesn't exactly match the depicted Standardese, but avoids needing an out-of-line function definition, and results in fewer member function calls), because the integral conversions concerned us. We concluded that this code has well-defined behavior and does not trigger arithmetic overflow. However, @mnatsuhara suggests adding a test case to cover this. The extreme scenario is storing a `year` of `-30'000` and then adding `60'000` `years`, resulting in year `30'000`. Similarly for the `operator-=`.